src/pkg/runner/runner.go

```diff
@@ -168,14 +186,49 @@ func (r *Runner) getTask(taskName string) (types.Task, error) {
 	return types.Task{}, fmt.Errorf("task name %s not found", taskName)
 }
 
+func mergeEnv(env1, env2 []string) []string {
+	for _, s1 := range env1 {
+		replaced := false
+		for j, s2 := range env2 {
+			if strings.Split(s1, "=")[0] == strings.Split(s2, "=")[0] {
+				env2[j] = s1
+				replaced = true
+			}
+		}
+		if !replaced {
+			env2 = append(env2, s1)
+		}
+	}
+	return env2
+}
+
+func formatEnvVar(name, value string) string {
+	// replace all non-alphanumeric characters with underscores
+	name = regexp.MustCompile(`[^a-zA-Z0-9]+`).ReplaceAllString(name, "_")
+	name = strings.ToUpper(name)
+	// prefix with INPUT_ (same as GitHub Actions)
+	return fmt.Sprintf("INPUT_%s=%s", name, value)
+}
+
 func (r *Runner) executeTask(task types.Task) error {
 	if len(task.Files) > 0 {
 		if err := r.placeFiles(task.Files); err != nil {
 			return err
 		}
 	}
 
+	defaultEnv := []string{}
+	for name, ip := range task.Inputs {
```

Comment:
nit: `ip` is Internet Protocol, rename to something that isn't so overloaded (`p` for param, spell out `param` or `inputParam`, etc)